tests/cases/compiler/controlFlowArrays.ts

```diff
@@ -0,0 +1,159 @@
+// @strictNullChecks: true
+// @noImplicitAny: true
+
+declare function cond(): boolean;
+
+function f1() {
+    let x = [];
+    x[0] = 5;
+    x[1] = "hello";
+    x[2] = true;
+    return x;  // (string | number | boolean)[]
+}

```

Comment:
Yes, you'd get the same type. We don't look at the value of the index expression, so nothing gained by an additional test.